### ISSUE TITLE
SFTP ls retrieves always detailed listing

### ISSUE THREAD

**Issue by juliospain** [intake/filesystem_spec#570]
The ls implementation for SFTP always retrieves the detailed info of each file or directory, just to throw it out if detailed was not needed.

```
    def ls(self, path, detail=False):
        out = ["/".join([path.rstrip("/"), p]) for p in self.ftp.listdir(path)]
        out = [self.info(o) for o in out]
        if detail:
            return out
        return sorted([p["name"] for p in out])
```

I have not tried it but something like this would make more sense:

```
    def ls(self, path, detail=False):
        paths = ["/".join([path.rstrip("/"), p]) for p in self.ftp.listdir(path)]
        if detail:
            return [self.info(p) for p in paths]
        else:
            return sorted(paths)
```

**Comment by martindurant** [comment]
Sounds reasonable! The only reason to catch the info every time, is if we cache it - but we currently don'e

**Comment by juliospain** [comment]
Funny thing, looking at the paramiko implementation and details, listdir is retrieving ALL the entries attributes and then returning only the names.

The entry details can be obtained using "listdir_attr" in a very efficient way. This takes in my use case around 370ms as opposed to the current implementation entry by entry 14seconds....

I'll work a PR out of this.

**Comment by juliospain** [comment]
Closed with #571
